### PR TITLE
Improve Prometheus example

### PR DESCRIPTION
**Pull request by reyang**
Removed the timeout (it's buggy, and doesn't have obvious benefit).
Used a real world async counter example.
Simplified the code.

### CODE REVIEW

File: examples/Console/TestPrometheusExporter.cs [hunk 1]
```diff
@@ -16,80 +16,63 @@
...
-    internal class TestPrometheusExporter
-    {
-        private static readonly Meter MyMeter = new Meter("TestMeter");
-        private static readonly Counter<double> Counter = MyMeter.CreateCounter<double>("myCounter", description: "A counter for demonstration purpose.");
-        private static readonly Histogram<long> MyHistogram = MyMeter.CreateHistogram<long>("myHistogram");
-        private static readonly ThreadLocal<Random> ThreadLocalRandom = new ThreadLocal<Random>(() => new Random());
-
-        internal static object Run(int port, int totalDurationInMins)
-        {
-            /*
```
Comment (cijothomas) [inline examples/Console/TestPrometheusExporter.cs:37]: i'd still keep a known prometheus.yml for quickly testing.

We include it in almost all examples in this folder
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/Console/TestOtlpExporter.cs#L30

File: examples/Console/TestPrometheusExporter.cs [hunk 2]
```diff
@@ -16,80 +16,63 @@
...
-            Following is sample prometheus.yml config. Adjust port,interval as needed.
+namespace Examples.Console;
 
-            scrape_configs:
-              # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
-              - job_name: 'OpenTelemetryTest'
+internal class TestPrometheusExporter
+{
+    private static readonly Meter MyMeter = new Meter("MyMeter");
+    private static readonly Counter<double> Counter = MyMeter.CreateCounter<double>("myCounter", description: "A counter for demonstration purpose.");
```
Comment (cijothomas) [inline examples/Console/TestPrometheusExporter.cs:32]: please put `unit` as well, so as to demo how it'll appear in prometheus.

**Review by cijothomas — APPROVED** [review]
LGTM.
Suggest to retain a valid prometheus.yml example in the code comment.

File: examples/Console/TestPrometheusExporter.cs (hunk 1 above)
Comment (reyang) [reply examples/Console/TestPrometheusExporter.cs:37]: I've added a simple yml file.

Curious, do we treat this project as an "example", a "test" or both?

File: examples/Console/TestPrometheusExporter.cs (hunk 2 above)
Comment (reyang) [reply examples/Console/TestPrometheusExporter.cs:32]: the thread runaway time has unit so I intend to not put unit here

File: examples/Console/TestPrometheusExporter.cs (hunk 1 above)
Comment (cijothomas) [reply examples/Console/TestPrometheusExporter.cs:37]: yes its a mix of both now. Agree we need a readme with instructions on how to run etc.

(I use them a lot, (and I believe other contributors as well), to quickly test things out.)

**Pull request merged**